packages/visualizations-react/rollup.config.js

```diff
@@ -0,0 +1,107 @@
+import postcss from 'rollup-plugin-postcss';
+import autoprefixer from 'autoprefixer';
+import visualizer from 'rollup-plugin-visualizer';
+import { terser } from 'rollup-plugin-terser';
+import typescript from '@rollup/plugin-typescript';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
+import json from '@rollup/plugin-json';
+import { babel } from '@rollup/plugin-babel';
+import replace from '@rollup/plugin-replace';
+import { defineConfig } from 'rollup';
+import pkg from './package.json';
+
+const production = !process.env.ROLLUP_WATCH;
+
+function basePlugins() {
+    return [
+        typescript({
+            sourceMap: true,
+            declaration: true,
+            declarationDir: 'dist',
+            rootDirs: ['src', 'test', 'stories'],
+        }),
+        nodeResolve(),
+        commonjs(),
+        json(),
+        postcss({
+            extract: 'index.css',
+            plugins: [autoprefixer()],
+        }),
+        // Transpile to ES5 when running a production build
+        production &&
+            babel({
+                babelHelpers: 'bundled',
+                extensions: ['.ts', '.mjs', '.js', '.tsx', 'jsx'],
+                include: ['src/**'],
+                presets: ['@babel/preset-env'],
+            }),
+    ];
+}
+
+function onwarn(warning, warn) {
+    // https://github.com/moment/luxon/issues/193
+    if (warning.code === 'CIRCULAR_DEPENDENCY') {
+        if (warning.importer.includes('node_modules/luxon')) {
+            return;
+        }
+    }
+    warn(warning);
+}
+
+const esm = defineConfig({
+    input: 'src/index.tsx',
+    // Externalize all dependencies
+    external: (id) => Object.keys(pkg.dependencies).includes(id),
+    output: {
+        dir: 'dist',
+        entryFileNames: '[name].es.js',
+        format: 'es',
+        sourcemap: true,
+    },
+    plugins: [
+        ...basePlugins(),
+        // Visualize the generated bundle
+        production &&
+            visualizer({
+                filename: `gen/stats-es.html`,
+                sourcemap: true,
+            }),
+    ],
+    onwarn,
+});
+
+const umd = defineConfig({
+    input: 'src/index.tsx',
+    output: {
+        dir: 'dist',
+        entryFileNames: '[name].umd.js',
+        format: 'umd',
+        sourcemap: true,
+        name: 'opendatasoft.visualizations',
```

Comment:
I'm actually wondering if an UMD build makes sense for a React library. For either way it should expose a different name to avoid conflicts here, such as `opendatasoft.visualizationsReact` (low-effort name 😅)

Reply:
For something like observable maybe? I'm not sure either… I don't see a reason to bundle to ES since we can import already. Maybe we don't bundle at all after all 😅 Same question could be asked for visualization…

Reply:
`visualization` can be used in contexts such as browsers by just using a script tag and running `new Chart()`, so UMD makes sense here IMO (also, Observable indeed). But `visualization-react` requires `React` and I'm not sure you can do a `React` app using script tags, or in Observable. But let's keep it for now I guess...